Add JunOS MIBs

loganmc10 opened the pull request:
Add vendor provided JunOS MIBs. Fixes https://github.com/prometheus/snmp_exporter/issues/578

brian-brazil commented:
This should include a generator.yml change, no point in having MIBs we don't use.

loganmc10 commented:
Isn't the idea that users can write their own generator.yml to fit their needs? Or do you mean just to include a few things as an example/base for people?

brian-brazil commented:
That's the general idea of the generator. The MIBs referenced here are used for the example generator.yml - and if a MIB isn't used there there's no point in downloading or processing it. So the goal of a PR like this should be to add some useful Juniper-specific modules.

loganmc10 commented:
I added an example for gathering the JunOS alarm values. I re-ran the generator and replaced the included snmp.yml file. Is this generally the right process? As you can see it removed a number of enum values when I did that

brian-brazil commented:
That's the right process, I'm guessing the MIBs you included include an older version of something so we'll need to be more judicious in what we use from Juniper.

The examples should also be reasonably usable as a starting point, so a bit more than just two metrics.

loganmc10 commented:
Ok I was more picky about what gets pulled in from Junipers "StandardMibs" folder and that solved the problem. I added a few more metrics to the generator for JunOS. I think just having the MIBs available like this makes it a lot better for JunOS users. All they need to do is all the OIDs they want, without needing to worry about importing the MIBs the right way.

I guess what I'm saying is that I think there is value in having MIBs in this repository even if the example generator.yml is fairly sparse. Cisco users already had this benefit since the Cisco MIBs are already there.

brian-brazil commented on generator/Makefile:
```diff
@@ -247,3 +250,25 @@ $(MIBDIR)/LIEBERT_GP_PDU.MIB:
+	@curl $(CURL_OPTS) -o $(TMP) $(JUNOS_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) "JuniperMibs/*" \
+                                      "StandardMibs/mib-rtmib.txt" \
+                                      "StandardMibs/mib-rfc*.txt" \
```
I'd expect that we'd already have these from Cisco, keep this minimal.

brian-brazil commented on generator/generator.yml:
```diff
@@ -503,3 +503,13 @@ modules:
+    walk:
+      - sysUpTime
+      - interfaces
+      - 1.3.6.1.4.1.2636.3.4.2.3.1 #Red Alarm state
```
This is still only 2 metrics. These examples should be something that someone would plausibly use in production.

brian-brazil reviewed (commented):
> I guess what I'm saying is that I think there is value in having MIBs in this repository even if the example generator.yml is fairly sparse.

Users can already download them themselves, they've not much use just being a URL in a file.

loganmc10 commented:
Ok thanks I'll just fork the repo
